src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -502,3 +506,76 @@ export const updateForm = (
     return transformMongoError(error)
   })
 }
+
+/**
+ * Updates form settings.
+ * @param originalForm The original form to update settings for
+ * @param body the subset of form settings to update
+ * @returns ok(updated form settings) on success
+ * @returns err(MalformedParametersError) if email update is attempted for an encrypt mode form
+ * @returns err(database errors) if db error is thrown during form setting update
+ */
+export const updateFormSettings = (
+  originalForm: IPopulatedForm,
+  body: SettingsUpdateDto,
+): ResultAsync<
+  FormSettings,
+  | DatabaseError
+  | DatabaseValidationError
+  | DatabaseConflictError
+  | DatabasePayloadSizeError
+  | MalformedParametersError
+> => {
+  // If non-empty string or even if any array is passed in, return with error if
+  // form is encrypt mode.
+  if (originalForm.responseMode === ResponseMode.Encrypt && !!body.emails) {
+    logger.error({
+      message: 'Attempted to update form emails on an encrypt mode form',
+      meta: {
+        action: 'updateFormSettings',
+        originalFormId: originalForm._id,
```

Comment:
should this just be `formId`, since the id will never change? just for consistency with the rest of the logs